### PR TITLE
Resolve S3 bucket for AWS::ApiGatewayV2::Api resource

### PR DESCRIPTION
**Pull request by wchengru**
#### Which issue(s) does this change fix?
#3459

#### Why is this change necessary?
When `BodyS3Location` of `AWS::ApiGatewayV2::Api` is set as a local file path, it is not resolved to a s3 location when running `sam package`.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/lib/package/packageable_resources.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
     METADATA_WITH_LOCAL_PATHS,
     RESOURCES_WITH_LOCAL_PATHS,
     RESOURCES_WITH_IMAGE_COMPONENT,
-    AWS_ECR_REPOSITORY,
+    AWS_ECR_REPOSITORY, AWS_APIGATEWAY_V2_API,
```
Comment (CoshUS) [inline samcli/lib/package/packageable_resources.py:49]:
```suggestion
    AWS_ECR_REPOSITORY, 
    AWS_APIGATEWAY_V2_API,
```

**Review by CoshUS — APPROVED** [review]

**Review by jfuss — COMMENTED** [review]
Do we need `sam sync` integ tests too? I noticed that this links to a `sam sync` issue so want to double check before approving.

**Comment by CoshUS** [comment]
> Do we need `sam sync` integ tests too? I noticed that this links to a `sam sync` issue so want to double check before approving.

`sam sync` integration tests are still in the process of being added. Although the initial issue mentions sam sync, sam sync only directly calls PackageContext and has no extra logic for infrastructure sync. 

https://github.com/aws/aws-sam-cli/pull/3495
@qingchm Is this something we can add for the infra integration tests or better suited for code sync ones?

**Comment by jfuss** [comment]
@CoshUS Got it. I think we need to get out of the habit of writing integ tests after the fact. I have noticed this in a couple different PRs. We can take that discussion within the team though.

**Review by jfuss — APPROVED** [review]

**Comment by qingchm** [comment]
> > Do we need `sam sync` integ tests too? I noticed that this links to a `sam sync` issue so want to double check before approving.
> 
> `sam sync` integration tests are still in the process of being added. Although the initial issue mentions sam sync, sam sync only directly calls PackageContext and has no extra logic for infrastructure sync.
> 
> #3495 @qingchm Is this something we can add for the infra integration tests or is it better suited for code sync ones?

I think this is more related to infra, would put this into our backlog and add tests for such cases later. Right now the sync integ tests are still in limited scope with working http v2 examples and I wouldn't put it in the current PR yet.

**Pull request merged**